gaffer-core/serialisation/src/main/java/gaffer/jsonserialisation/JSONSerialiser.java

```diff
@@ -124,31 +126,70 @@ public void serialise(final Object object, final JsonGenerator jsonGenerator, fi
     }
 
     /**
+     * Used to convert a {@link Class} into a {@link com.fasterxml.jackson.core.type.TypeReference}
+     * @param <T> the class type
+     */
+    private static class ClassTypeReference<T> extends TypeReference<T> {
+        private final Class<T> clazz;
+
+        ClassTypeReference(final Class<T> clazz) {
+            this.clazz = clazz;
+        }
+
+        @Override
+        public Type getType() {
+            return clazz;
+        }
+    }
+
+    /**
      * @param bytes the bytes of the object to deserialise
      * @param clazz the class of the object to deserialise
      * @param <T>   the type of the object
      * @return the deserialised object
      * @throws SerialisationException if the bytes fail to deserialise
      */
     public <T> T deserialise(final byte[] bytes, final Class<T> clazz) throws SerialisationException {
+        return deserialise(bytes, new ClassTypeReference<>(clazz));
+    }
+
+    /**
+     * @param stream the {@link java.io.InputStream} containing the bytes of the object to deserialise
+     * @param clazz   the class of the object to deserialise
+     * @param <T>    the type of the object
+     * @return the deserialised object
+     * @throws SerialisationException if the bytes fail to deserialise
+     */
+    public <T> T deserialise(final InputStream stream, final Class<T> clazz) throws SerialisationException {
+        return deserialise(stream, new ClassTypeReference<>(clazz));
```

Comment:
These deserialise methods are used a lot, so we should minimise the number of objects we instantiate. Although it is more code, it might be better if 

```java
deserialise(final byte[] bytes, final Class<T> clazz)
```

just delegated directly to mapper.readValue(bytes, clazz);